test/acceptance_tests/test_eeg_classifier.py

```diff
@@ -188,82 +188,66 @@ def test_eeg_classifier():
+        {'batches': [{'train_batch_size': 32, 'train_loss': 1.9391239881515503},
+                     {'train_batch_size': 32, 'train_loss': 2.895704507827759},
+                     {'train_batch_size': 32, 'train_loss': 1.0713887214660645},
+                     {'valid_batch_size': 24, 'valid_loss': 1.18110191822052}],
+         'epoch': 1,
+         'train_batch_count': 3,
+         'train_loss': 1.9687390724817913,
+         'train_loss_best': True,
+         'train_trial_accuracy': 0.4791666666666667,
+         'train_trial_accuracy_best': True,
+         'valid_batch_count': 1,
+         'valid_loss': 1.18110191822052,
+         'valid_loss_best': True,
+         'valid_trial_accuracy': 0.5,
+         'valid_trial_accuracy_best': True},
+        {'batches': [{'train_batch_size': 32, 'train_loss': 1.6741573810577393},
+                     {'train_batch_size': 32, 'train_loss': 0.9984264373779297},
+                     {'train_batch_size': 32, 'train_loss': 1.1340471506118774},
+                     {'valid_batch_size': 24, 'valid_loss': 2.5375664234161377}],
+         'epoch': 2,
+         'train_batch_count': 3,
+         'train_loss': 1.2688769896825154,
+         'train_loss_best': True,
+         'train_trial_accuracy': 0.5,
+         'train_trial_accuracy_best': True,
+         'valid_batch_count': 1,
+         'valid_loss': 2.5375664234161377,
+         'valid_loss_best': False,
+         'valid_trial_accuracy': 0.5,
+         'valid_trial_accuracy_best': False},
+        {'batches': [{'train_batch_size': 32, 'train_loss': 0.8795645833015442},
+                     {'train_batch_size': 32, 'train_loss': 1.0339491367340088},
+                     {'train_batch_size': 32, 'train_loss': 1.19275963306427},
+                     {'valid_batch_size': 24, 'valid_loss': 1.655737042427063}],
+         'epoch': 3,
+         'train_batch_count': 3,
+         'train_loss': 1.0354244510332744,
+         'train_loss_best': True,
+         'train_trial_accuracy': 0.5,
+         'train_trial_accuracy_best': False,
+         'valid_batch_count': 1,
+         'valid_loss': 1.655737042427063,
+         'valid_loss_best': False,
+         'valid_trial_accuracy': 0.5,
+         'valid_trial_accuracy_best': False},
+        {'batches': [{'train_batch_size': 32, 'train_loss': 1.1963350772857666},
+                     {'train_batch_size': 32, 'train_loss': 0.8621770143508911},
+                     {'train_batch_size': 32, 'train_loss': 1.099318265914917},
+                     {'valid_batch_size': 24, 'valid_loss': 1.0293445587158203}],
+         'epoch': 4,
+         'train_batch_count': 3,
+         'train_loss': 1.0526101191838582,
+         'train_loss_best': False,
+         'train_trial_accuracy': 0.625,
+         'train_trial_accuracy_best': True,
+         'valid_batch_count': 1,
+         'valid_loss': 1.0293445587158203,
+         'valid_loss_best': True,
+         'valid_trial_accuracy': 0.25,
+         'valid_trial_accuracy_best': False}
```

Comment:
you bascially updated all the tests @robintibor so we cannot easily assess that your PR is a bug fix.

Reply:
The reason is that I now changed behavior to enforce that adding metrics will not change random state. Before, the results would actually change depending on if you had any extra metric added or none, because this would affect the random state during training.